aws/image/misc-settings.sh

```diff
@@ -38,85 +38,37 @@ then
     esac
 fi
 
-if [[ "$CLOUD_PROVIDER" == "azure" && "$PODVM_DISTRO" == "ubuntu" ]]; then
+if [[ "$CLOUD_PROVIDER" == "azure" || "$CLOUD_PROVIDER" == "generic" ]] && [[ "$PODVM_DISTRO" == "ubuntu" ]]; then
```

Comment:
In this case I would also consider removing the iptables check and install it for every distro.

Reply:
We don't have a target for it but we pass "generic" as the value for CLOUD_PROVIDER in the [workflow](https://github.com/confidential-containers/cloud-api-adaptor/blob/4a6d01090e808325bfa7b366cc1def43721f9d35/.github/workflows/podvm.yaml#L60)

Reply:
Ok, I see. We use the `generic` in the workflow to add a prefix for the generated artefacts. But using `generic` might be confusing here. Or we'll have to add the description of `generic` cloud provider in the readme. Wdyt ?
What about removing the cloud provider check or just checking if it's not vsphere before adding the imds service ?

Reply:
If we are planning to have generic artifacts i think this would be the better option.
IIUC there's no problem with vSphere, the only reason it's separated is additional packages being installed (mabye we can install them for the generic use case? dunnu)

Reply:
vSphere may need more additional packages. Who knows. Better to keep it separate imho